### PR TITLE
[mp] quickstart rewrite

### PR DESCRIPTION
**Pull request by mattppal**
# Description

Rewrites the quickstarts to guide users through setup and install of Mage. This quickstart gives preference towards starting form docker, specifically the docker compose template, as it's most clear how Mage is being executed. Also included are instructions for running `example_pipeline` from the Mage UI.

There are a few other misc edits in here, too:
- Variables overview page
- OG:images
- small tweaks/fixes

Quickstart - Mage (1)

# How Has This Been Tested

- [ ] `mintlify dev`

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
Did you deploy this to staging so that I can preview?

**Comment by mattppal** [comment]
> Did you deploy this to staging so that I can preview?

https://mage-staging.mintlify.app/getting-started/setup

File: docs/development/variables/environment-variables.mdx [hunk 1]
```diff
@@ -4,49 +4,117 @@ sidebarTitle: "Environment variables"
...
+  mage:
+    image: mageai/mageai:latest
+    ...
+    volumes:
+      - ~/.aws:/root/.aws
+```
+
+## Default variables
+
+These are the environment variables set by default in Mage.
```
Comment (wangxiaoyou1993) [inline docs/development/variables/environment-variables.mdx:50]: `set by default in Mage` => `used in Mage`

File: docs/development/variables/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+description: "There are a few ways to set up variables in Mage, here's how you can use them to make the best pipelines the world has ever seen. 🌎"
+---
+
+There are four types of variable in Mage: Environment variables, runtime variables, block variables, and secrets.
+
+| Variable Type        | Scope    | Creation | Reference                       | Encrypted |
+| -------------------- | -------- | -------- | ------------------------------- | --------- |
+| Environment variable | System   | Code     | `{{ env_var() }}`               | 🚫       |
+| Runtime variable     | Pipeline | Code/UI  | `kwargs` or `{{ variables() }}` | 🚫       |
+| Block variable       | Block    | Code/UI  | `kwargs`                        | 🚫       |
```
Comment (wangxiaoyou1993) [inline docs/development/variables/overview.mdx:13]: `kwargs['configuration'][key]`

File: docs/development/variables/environment-variables.mdx [hunk 3]
```diff
@@ -4,49 +4,117 @@ sidebarTitle: "Environment variables"
...
+You can also configure the base path separately for client requests and backend routes by setting
+the `MAGE_REQUESTS_BASE_PATH` and `MAGE_ROUTES_BASE_PATH` environment variables respectively.
+
+|                           |                                                                                     |
+| ------------------------- | ----------------------------------------------------------------------------------- |
+| `MAGE_BASE_PATH`          | Set base path for the entire app.                                                   |
+| `MAGE_REQUESTS_BASE_PATH` | Set base path for frontend requests and urls, defaults to value of `MAGE_BASE_PATH` |
+| `MAGE_ROUTES_BASE_PATH`   | Set base path for backend routes, defaults to value of `MAGE_BASE_PATH`             |
+
+## Using credential files in Docker
```
[the remark this answers is not in the document]
Comment (mattppal) [reply docs/development/variables/environment-variables.mdx]: Fixed. I think this should go in a FAQ/troubleshooting section

File: docs/getting-started/setup.mdx [hunk 4]
```diff
@@ -1,440 +1,130 @@
...
-
-To use a specific schema in the PostgreSQL database, the connection string could contain the schema name as a parameter:
+   Be sure that [Docker](https://docs.docker.com/engine/install/) and [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) are installed and run:
 
    ```bash
-   docker run --network mage-app -it -p 6789:6789 -v $(pwd):/home/src \
-      -e MAGE_DATABASE_CONNECTION_URL="postgresql+psycopg2://<username>:<password>@postgres_db:5432/<database>?options=-c%%20search_path%%3D<schema_name>" \
-      mageai/mageai \
-      /app/run_app.sh mage start [project_name]
+   git clone https://github.com/mage-ai/compose-quickstart.git mage-quickstart \
```
Comment (wangxiaoyou1993) [inline docs/getting-started/setup.mdx:26]: One thing I'm not sure is whether it works on windows.

File: docs/getting-started/setup.mdx [hunk 5]
```diff
@@ -1,440 +1,130 @@
...
-    -v $(pwd):/home/src mageai/mageai \
-    -m 2048m \
-    /app/run_app.sh mage start [project_name]
-```
-
----
-
-## Install
-
-### Installing extra packages
```
Comment (wangxiaoyou1993) [inline docs/getting-started/setup.mdx:355]: Where do we put the instructions of installing extra packages?

File: docs/getting-started/setup.mdx [hunk 6]
```diff
@@ -1,440 +1,130 @@
...
-
-## Using secret credential files in Docker
-
-If you have files containing secret credentials (e.g. GCP credentials JSON files),
-you’ll need to mount them as a volume so that those files appear in your running Docker container.
-
-### macOS
+   </Accordion>
+   <Accordion title="Kubernetes" icon="network-wired">
+      1. Start a Kubernetes cluster locally
```
Comment (wangxiaoyou1993) [inline docs/getting-started/setup.mdx:92]: The number looks off on staging 
image

File: docs/getting-started/setup.mdx (hunk 4 above)
Comment (mattppal) [reply docs/getting-started/setup.mdx:26]: Those are all pretty universal commands (git clone, cd, cp, rm) so it should work perfectly.

File: docs/getting-started/setup.mdx (hunk 6 above)
Comment (mattppal) [reply docs/getting-started/setup.mdx:92]: Looks good on my end :) 

image

File: docs/getting-started/setup.mdx (hunk 4 above)
Comment (wangxiaoyou1993) [reply docs/getting-started/setup.mdx:26]: Some of these commands on different on windows: https://www.geeksforgeeks.org/linux-vs-windows-commands/

Comment (mattppal) [reply docs/getting-started/setup.mdx:26]: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/copy-item?view=powershell-7.3#notes

`cp` is aliased to `Copy-Item`

https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/remove-item?view=powershell-7.3#notes

`rm` is aliased to `Remove-Item`

`cd` is the same

Powershell does not use `&&`

```
git clone https://github.com/mage-ai/compose-quickstart.git mage-quickstart
   cd mage-quickstart
   cp dev.env .env
   rm dev.env
   docker compose up
```
updated

File: docs/getting-started/setup.mdx [hunk 7]
```diff
@@ -1,440 +1,164 @@
...
+   Once the server is running, open `http://localhost:6789` in your browser and explore! We recommend one of our [guides](/guides/overview) to for inspiration.
 
----
+   ### What's happening?
 
-## Download new version of Mage
+   We're executing `docker run`, which runs a Docker container from the Mage image. additionally, we're _mounting a volume_ (the `-v` command) to persist our project files on our container. This mounts the current folder to `home/src` in the mage container. We're executing our docker image `mageai/mageai` and running a script to start up the container.
 
-### Using Docker
+   If you'd like to connect to an external database, either Postgres or MSSQL, using Docker containers, please see the [external database quickstart](/getting-started/external-database-quickstart).
```
Comment (wangxiaoyou1993) [inline docs/getting-started/setup.mdx:64]: Where is the doc for "/getting-started/external-database-quickstart"?

File: docs/getting-started/setup.mdx [hunk 8]
```diff
@@ -1,440 +1,164 @@
...
+   </CodeGroup>
 
-   If you want the pipeline blocks to be executed in separate containers, you can follow the [guide](/production/configuring-production-settings/compute-resource#kubernetes-executor) to set `executor_type` of a block to `k8s` in pipeline's `metadata.yaml`.
+   Once the server is running, open `http://localhost:6789` in your browser and explore! We recommend one of our [guides](/guides/overview) to for inspiration.
 
----
+   ### What's happening?
 
-## Download new version of Mage
+   We're executing `docker run`, which runs a Docker container from the Mage image. additionally, we're _mounting a volume_ (the `-v` command) to persist our project files on our container. This mounts the current folder to `home/src` in the mage container. We're executing our docker image `mageai/mageai` and running a script to start up the container.
```
Comment (wangxiaoyou1993) [inline docs/getting-started/setup.mdx:62]: nit: `home/src` => `/home/src`

File: docs/getting-started/setup.mdx (hunk 5 above)
Comment (wangxiaoyou1993) [reply docs/getting-started/setup.mdx:355]: Did you address this comment?

File: docs/getting-started/setup.mdx (hunk 7 above)
Comment (mattppal) [reply docs/getting-started/setup.mdx]: I removed this reference. This is more of a docker tutorial than a mage quickstart

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**